build.rs

```diff
@@ -34,6 +35,33 @@ fn main() {
         // so for those we want to apply the `#[allow(unused_unsafe)]` attribute to get rid of that warning.
         println!("cargo:rustc-cfg=allow_unused_unsafe");
     }
+
+    let target_arch = match std::env::var("CARGO_CFG_TARGET_ARCH") {
+        Ok(arch) => arch,
+        _ => "".to_string(),
+    };
+
+    // Backend override
+    let curve25519_dalek_backend = match std::env::var("CARGO_CFG_CURVE25519_DALEK_BACKEND")
+        .as_deref()
+    {
+        Ok("fiat") => "fiat",
+        Ok("serial") => "serial",
+        Ok("simd") => {
+            // simd override is not guaranteed as:
+            // simd can only be enabled on x86_64 & 64bit target_pointer_width
+            if target_arch == "x86_64" && curve25519_dalek_bits == DalekBits::Dalek64 {
+                "simd"
+            // fallback to serial with a warning
+            } else {
+                println!("cargo:warning=Could not override curve25519_dalek_backend to simd - defaulting to serial");
+                "serial"
+            }
+        }
+        // default serial
+        _ => "serial",
```

Comment:
Won't this *disable* SIMD by default unless it's enabled by `CARGO_CFG_CURVE25519_DALEK_BACKEND`? Or am I reading this wrong?

Reply:
Oops - yeah my last commit was brainfreeze clearly - fixed I think

Reply:
@koute I think if if someone tries to override backend to "simd" whilst only "serial" is possible we should do a compile error isntead - this would enable to simplify the gates a lot in backend mod.rs where we don't need a have a litany of all(not()'s